integrations/kube-agent-updater/pkg/version/basichttp.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/cache"
+)
+
+const (
+	// versionPath is the version discovery endpoint returning the current
+	// target version as defined in
+	// https://github.com/gravitational/teleport/blob/master/rfd/0109-cloud-agent-upgrades.md#version-discovery-endpoint
+	versionPath   = "version"
+	httpTimeout   = 10 * time.Second
+	cacheDuration = time.Minute
```

Comment:
Nit: Should we move these constants and constants from "maintenance" package to a common "constants" file? Then we could also reuse timeout/duration in both places, I don't imagine those should necessarily be different?